File: spec/php/ProcessXor4ValueTest.php
<?php
namespace Kaitai\Struct\Tests;

class ProcessXor4ValueTest extends TestCase {
    public function testProcessXor4Value() {
        $r = ProcessXor4Value::fromFile(self::SRC_DIR_PATH . "/process_xor_4.bin");

        $this->assertEquals("\xec\xbb\xa3\x14", $r->key());
        $this->assertEquals("foo bar", $r->buf());
    }
}

File: spec/go/process_xor4_value_test.go
package spec

import (
	"os"
	"testing"

	"github.com/kaitai-io/kaitai_struct_go_runtime/kaitai"
	"github.com/stretchr/testify/assert"

	. "test_formats"
)

func TestProcessXor4Value(t *testing.T) {
	f, err := os.Open("../../src/process_xor_4.bin")
	if err != nil {
		t.Fatal(err)
	}
	s := kaitai.NewStream(f)

	var r ProcessXor4Value
	err = r.Read(s, &r, &r)
	if err != nil {
		t.Fatal(err)
	}

	assert.EqualValues(t, []byte{0xec, 0xbb, 0xa3, 0x14}, r.Key)
	assert.EqualValues(t, "foo bar", r.Buf)
}
